pontoon/base/models/entity.py

```diff
@@ -843,24 +845,60 @@ def for_project_locale(
         if search:
             search_list = utils.get_search_phrases(search)
 
-            translation_filters = (
-                Q(translation__string__icontains_collate=(search, locale.db_collation))
-                & Q(translation__locale=locale)
-                for search in search_list
-            )
-            translation_matches = entities.filter(*translation_filters).values_list(
-                "id", flat=True
-            )
+            if search_identifiers:
+                translation_filters = (
+                    Q(
+                        translation__string__icontains_collate=(
+                            search,
+                            locale.db_collation,
+                        )
+                    )
+                    & Q(translation__locale=locale)
+                    for search in search_list
+                )
+                translation_matches = entities.filter(*translation_filters).values_list(
+                    "id", flat=True
+                )
+            else:
+                entities = entities.annotate(
+                    filtered_string=Substr("string", StrIndex("string", Value("=")) + 1)
+                )
+                translation_filters = (
+                    Q(filtered_string__icontains_collate=(search, locale.db_collation))
+                    & Q(translation__locale=locale)
+                    for search in search_list
+                )
+                translation_matches = entities.filter(*translation_filters).values_list(
+                    "id", flat=True
+                )
 
-            entity_filters = (
-                Q(string__icontains=search)
-                | Q(string_plural__icontains=search)
-                | Q(key__icontains=search)
-                for search in search_list
-            )
-            entity_matches = entities.filter(*entity_filters).values_list(
-                "id", flat=True
-            )
+            q_key = Q(key__icontains=search) if search_identifiers else Q()
+
+            if not search_identifiers:
+                entities = entities.annotate(
+                    filtered_string=Substr("string", StrIndex("string", Value("=")) + 1)
+                )
+                entity_filters = (
+                    Q(filtered_string__icontains=search) for search in search_list
+                )
+            else:
+                entity_filters = (
+                    Q(string__icontains=search)
+                    | Q(string_plural__icontains=search)
+                    | q_key
```

Comment:
You can drop `q_key` now:
```suggestion
                    | Q(key__icontains=search)
```